melodies_monet/driver.py

```diff
@@ -1156,7 +1176,22 @@ def plotting(self):
                                 vmax = None
                             # Select time to use as index.
                             pairdf = pairdf.set_index(grp_dict['data_proc']['ts_select_time'])
-                            a_w = grp_dict['data_proc']['ts_avg_window']
+                            # Specify ts_avg_window if noted in yaml file. #qzr++
+                            if 'ts_avg_window' in grp_dict['data_proc'].keys():
+                                a_w = grp_dict['data_proc']['ts_avg_window']
+                            else:
+                                a_w = None
+
+                            #Steps needed to subset if secondary y-axis (altitude_variable) limits are provided, as make_timeseries from surfaceplots.py already plots the whole df
+                            # Extract altitude (secondary y-axis) limits (if provided)
+                            vmin2 = grp_dict['data_proc'].get('vmin2', None)
+                            vmax2 = grp_dict['data_proc'].get('vmax2', None)
+                            # Subset the data based on vmin2 and vmax2 if provided
+                            if vmin2 is not None and vmax2 is not None:
+                                altitude_variable = grp_dict['data_proc']['altitude_variable']
+                                pairdf = pairdf[(pairdf[altitude_variable] >= vmin2) & (pairdf[altitude_variable] <= vmax2)]
+
```

Comment:
I think that the use of vmin2 and vmax2 should be consistent with vmin_plot and vmax_plot for the primary axis, where it uses vmin_plot and vmax_plot to set the ylim of the primary axis. The same subsetting functionality here can be achieved with either the filter_string and filter_dict yaml options and it could be confusing to the user if the functionality of vmin and vmax is different throughout. Also setting altitude_variable here is redundant because it is set below.